From 54f57baab644e99b6da34d9538b2a9c0a05b690d Mon Sep 17 00:00:00 2001
From: Paul Elder <paul.elder@pitt.edu>
Date: Fri, 16 Jun 2017 00:54:20 +0900
Subject: [PATCH 26/41] kselftest: make ksft_exit_skip() output a reason for
 skipping

Make ksft_exit_skip() input an optional message string as the reason
for skipping all the tests and outputs it prior to exiting.

Signed-off-by: Paul Elder <paul.elder@pitt.edu>
Signed-off-by: Shuah Khan <shuahkh@osg.samsung.com>
---
 tools/testing/selftests/kselftest.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/tools/testing/selftests/kselftest.h b/tools/testing/selftests/kselftest.h
index 1d874a50d957..be01f2d15472 100644
--- a/tools/testing/selftests/kselftest.h
+++ b/tools/testing/selftests/kselftest.h
@@ -104,9 +104,12 @@ static inline int ksft_exit_xpass(void)
 	exit(KSFT_XPASS);
 }
 
-static inline int ksft_exit_skip(void)
+static inline int ksft_exit_skip(const char *msg)
 {
-	ksft_print_cnts();
+	if (msg)
+		printf("1..%d # Skipped: %s\n", ksft_test_num(), msg);
+	else
+		ksft_print_cnts();
 	exit(KSFT_SKIP);
 }
 
-- 
2.13.0

